backend/store/principal.go

```diff
@@ -393,8 +409,10 @@ func (s *Store) UpdateUser(ctx context.Context, userID int, patch *UpdateUserMes
 		s.projectIDPolicyCache.Purge()
 		s.projectPolicyCache.Purge()
 	}
-	s.userIDCache.Remove(userID)
-	return s.GetUserByID(ctx, userID)
+	s.userEmailCache.Remove(currentUser.Email)
```

Comment:
Don't need this?

Reply:
Nice catch but yes intended. When user changes email, we need to revoke the previous email.